src/pipeline/rackhd/docker/publish.sh

```diff
@@ -0,0 +1,141 @@
+    echo "      --DOCKER_RECORD_STASH_PATH: The directory path of the docker build record file."
+}
+
+###################################################################
+#
+#  Parse and check Arguments
+#
+##################################################################
+parseArguments(){
+    while [ "$1" != "" ]; do
+        case $1 in
+            --DOCKER_STASH_PATH )           shift
+                                            DOCKER_STASH_PATH=$1
+                                            ;;
+            --DOCKERHUB_USER )              shift
+                                            DOCKERHUB_USER=$1
+                                            ;;
+            --DOCKERHUB_PASS )              shift
+                                            DOCKERHUB_PASS=$1
+                                            ;;
+            --DOCKER_RECORD_STASH_PATH )    shift
+                                            DOCKER_RECORD_STASH_PATH=$1
+                                            ;;
+            * )
+                                            Usage
+                                            exit 1
+        esac
+        shift
+    done
+    if [ ! -n "${DOCKER_STASH_PATH}" ]; then
+        echo "[Error]Arguments DOCKER_STASH_PATH is required"
+        exit 1
+    fi
+    if [ ! -n "${DOCKERHUB_USER}" ]; then
+        echo "[Error]Arguments DOCKERHUB_USER is required"
+        exit 1
+    fi
+    if [ ! -n "${DOCKERHUB_PASS}" ]; then
+        echo "[Error]Arguments DOCKERHUB_PASS is required"
+        exit 1
+    fi
+    if [ ! -n "${DOCKER_RECORD_STASH_PATH}" ]; then
+        echo "[Error]Arguments DOCKER_RECORD_STASH_PATH is required"
+        exit 1
+    fi
+}
+
+# Clean UP. (was in Jenkins job post-build, avoid failure impacts build status.)
+cleanUp(){
+    # parameter : the images keyword to be delete
+    keyword=$1
+    images=`docker images | grep ${keyword} | awk '{print $3}' | sort | uniq`
+    if [ -n "$images" ]; then
+        docker rmi -f $images
+    fi
+}
+
+cleanRunningContainers() {
+    local containers=$(docker ps -a -q)
+    if [ "$containers" != "" ]; then
+        echo "Clean Up containers : " ${containers}
+        docker stop ${containers}
+        docker rm  ${containers}
+    fi
+}
+
+########################################################
+#
+# Main
+#
+########################################################
+
+main(){
+    echo "docker publish start to parse arguments."
+    parseArguments "$@"
+
+    docker load -i $DOCKER_STASH_PATH
+    docker login -u $DOCKERHUB_USER -p $DOCKERHUB_PASS
+
+    while read -r LINE; do
+            echo $LINE
+            for repo_tag in $LINE; do
+                    echo "Pushing rackhd/$repo_tag"
+            docker push $repo_tag
+            if [ $? != 0 ]; then
+                    echo "Failed to push rackhd/$repo_tag"
+                exit 1
+            fi
+        done
+    done < $DOCKER_RECORD_STASH_PATH
```

Comment:
The tag info is the output of ```docker load```
Could you please remove the parameter "DOCKER_RECORD_STASH_PATH"?

Reply:
Ok. That means `docker push` command will push all repos in the docker tar file  to the hub.docker.com